File: OOP/3_private_encaptulaiton.cpp
#include <iostream>
#include <string>

using namespace std;

class Person {
private:
    // Private attributes
    string name;
    int age;

    // Private method
    void display_info() {
        cout << name << " - " << age << endl;
    }

public:
    // Constructor to initialize private attributes
    Person(string name, int age) {
        this->name = name;
        this->age = age;
    }

    // Public method to access the private method
    void show_info() {
        display_info();
    }
};

int main() {
    // Create an object of the Person class
    Person p("Salim", 28);

    // Trying to access private method directly (this will raise an error in C++ if attempted)
    // p.display_info(); // Error: 'display_info' is a private member of 'Person'

    // Access private method through public method
    p.show_info();  

    return 0;
}


// Output: Salim - 28

File: OOP/3_private_encaptulaiton.py
class person:
    def __init__(self,name,age) -> None:
        self.__name = name
        self.__age = age   # private attribute

    def __display_info(self):  #private method
        print(self.__name,self.__age)

    # Public method to access private attributes
    def show_info(self):
        self.__display_info()


p = person("salim",28)
# p.__display_info()  will raise attribue error cause private methos can not be accessed outside from the class
p.show_info() # but private method can be  accessed through public method

